File: frontend/src/types/queue.ts
export type QueueLevel = "leve" | "moderada" | "intensa";

export interface WeightedPercents {
  leve: number;
  moderada: number;
  intensa: number;
}

export interface QueueStatusResponse {
  status: QueueLevel | null;
  hasEnoughData: boolean;
  totalVotes: number;
  votesNeeded: number;
  weightedPercents: WeightedPercents;
}

export interface PeakHourSlot {
  slot: string;
  avgIntensity: number | null;
  sampleCount: number;
  noData: boolean;
}

export interface TrendResponse {
  points: number[];
  trendUp: boolean;
}

export type PeakHoursMeal = "lunch" | "dinner";

File: backend/schemas/queue.py
from enum import Enum

from pydantic import BaseModel, Field


class QueueLevel(str, Enum):
    LEVE = "leve"
    MODERADA = "moderada"
    INTENSA = "intensa"


class QueueVoteRequestSchema(BaseModel):
    level: QueueLevel = Field(..., description="Nível de lotação percebido na fila")


class WeightedPercents(BaseModel):
    leve: float = 0.0
    moderada: float = 0.0
    intensa: float = 0.0


class QueueStatusResponseSchema(BaseModel):
    status: QueueLevel | None = None
    hasEnoughData: bool
    totalVotes: int
    votesNeeded: int
    weightedPercents: WeightedPercents


class PeakHourSlotSchema(BaseModel):
    slot: str
    avgIntensity: float | None = None
    sampleCount: int
    noData: bool


class TrendResponseSchema(BaseModel):
    points: list[float]
    trendUp: bool
